sdk/go/common/resource/config/object.go

```diff
@@ -0,0 +1,487 @@
+		}
+		return NewSecurePlaintext(plaintext), nil
+	case []object:
+		vs := make([]Plaintext, len(v))
+		for i, v := range v {
+			pv, err := v.Decrypt(ctx, decrypter)
+			if err != nil {
+				return Plaintext{}, err
+			}
+			vs[i] = pv
+		}
+		return NewPlaintext(vs), nil
+	case map[string]object:
+		vs := make(map[string]Plaintext, len(v))
+		for k, v := range v {
+			pv, err := v.Decrypt(ctx, decrypter)
+			if err != nil {
+				return Plaintext{}, err
+			}
+			vs[k] = pv
+		}
+		return NewPlaintext(vs), nil
+	default:
+		contract.Failf("unexpected value of type %T", v)
+		return Plaintext{}, nil
+	}
+}
+
+// Get gets the member value at path. The path to the receiver is prefix.
+func (c object) Get(prefix, path resource.PropertyPath) (object, bool, error) {
+	if len(path) == 0 {
+		return c, true, nil
+	}
+
+	prefix = append(prefix, path[0])
+	switch v := c.value.(type) {
+	case []object:
+		index, ok := path[0].(int)
+		if !ok || index < 0 || index >= len(v) {
+			return object{}, false, nil
+		}
+		elem := v[index]
+		return elem.Get(prefix, path[1:])
+	case map[string]object:
+		key, ok := path[0].(string)
+		if !ok {
+			return object{}, false, nil
+		}
+		elem, ok := v[key]
+		if !ok {
+			return object{}, false, nil
+		}
+		return elem.Get(prefix, path[1:])
+	default:
+		return object{}, false, nil
+	}
+}
+
+// Delete deletes the member value at path. The path to the receiver is prefix.
+func (c *object) Delete(prefix, path resource.PropertyPath) error {
+	if len(path) == 0 {
+		return nil
+	}
+
+	prefix = append(prefix, path[0])
+	switch v := c.value.(type) {
+	case []object:
+		index, ok := path[0].(int)
+		if !ok || index < 0 || index >= len(v) {
+			return nil
+		}
+		if len(path) == 1 {
+			c.value = append(v[:index], v[index+1:]...)
+			return nil
+		}
+		elem := &v[index]
+		return elem.Delete(prefix, path[1:])
+	case map[string]object:
+		key, ok := path[0].(string)
+		if !ok {
+			return nil
+		}
+		if len(path) == 1 {
+			if len(v) == 2 {
+				keys := make([]string, 0, 2)
+				for k := range v {
+					if k != key {
+						keys = append(keys, k)
+					}
+				}
+				if len(keys) == 1 && keys[0] == "secure" {
+					if _, ok := v["secure"].value.(string); ok {
+						return errSecureReprReserved
+					}
+				}
+			}
+
+			delete(v, key)
+			return nil
+		}
+		elem, ok := v[key]
+		if !ok {
+			return nil
+		}
+		err := elem.Delete(prefix, path[1:])
+		v[key] = elem
+		return err
+	default:
+		return nil
+	}
+}
+
+func newContainer(accessor any) any {
+	switch accessor := accessor.(type) {
+	case int:
+		return make([]object, accessor+1)
+	case string:
+		return make(map[string]object)
+	default:
+		contract.Failf("unexpected accessor kind %T", accessor)
+		return nil
+	}
+}
+
+// Set sets the member value at path to new. The path to the receiver is prefix.
+func (c *object) Set(prefix, path resource.PropertyPath, new object) error {
+	if len(path) == 0 {
+		*c = new
+		return nil
+	}
+
+	switch c.value.(type) {
```

Comment:
This switch statement looks like it's trying to guarantee an invariant for the block that follows.
A comment above it explaining this would be useful.